src/main/java/io/knowledgeassets/myskills/server/security/MySkillsSecurityExpressionRoot.java

```diff
@@ -14,14 +13,14 @@
 	private Object filterObject;
 	private Object returnObject;
 	private Object target;
-	private UserPermissionQueryService userPermissionQueryService;
-	private CommunityQueryService communityQueryService;
+	private final UserPermissionQueryService userPermissionQueryService;
+	private final SecurityService securityService;
```

Comment:
I really like the idea of decoupling the `SecurityExpressionRoot` from the business logic which does the various security checks. I recommend to extract all logic to the `SecurityService`, so that the expression root has only this service as a dependency. Then the `SecurityService` can have multiple dependencies to other services like `CommunityQueryService` and `UserPermissionQueryService` to provide security information. What do you think?

Reply:
I consider extraction of all security logic and placing it in `SecurityService` is a good idea. At least that was my intention when working on the PR to do this in future. I suggest to do this in a separate PR in the scope of refactoring.

Reply:
Agreed. 👍 Let's do it separately.